sky/client/sdk.py

```diff
@@ -101,6 +102,22 @@ def check(clouds: Optional[Tuple[str]],
     Request Returns:
         None
     """
+    if infra_list is None:
+        clouds = None
+    else:
+        specified_clouds = []
+        for infra_str in infra_list:
+            infra = infra_utils.InfraInfo.from_str(infra_str)
+            if infra.cloud is None:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(f'Invalid infra to check: {infra_str}')
+            if infra.region is not None or infra.zone is not None:
+                region_zone = infra_str.partition('/')[-1]
+                logger.warning(f'Infra {infra_str} is specified, but `check` '
+                               f'only supports checking {infra.cloud}, '
+                               f'ignoring {region_zone}')
+            specified_clouds.append(infra.cloud)
+        clouds = tuple(specified_clouds)
```

Comment:
If the infra -> cloud/region/zone conversion is handled in cli.py for everything else, why do we handle it in sdk.py only for check?

Reply:
It's also handled in SDK by `Resources` class. It is just that we have some CLI logic that also requires those field to present.